File: sandbox/marissa/java/Node.java
public class Node<T extends Comparable<T>>
{
	private Node<T> next;
	private T element;

	public Node()
	{
		this.next = null;
		this.element = null;
	}

	public Node(T element)
	{
		this.next = null;
		this.element = element;
	}

	public void setNext(Node<T> next)
	{
		this.next = next;
	}

	public Node<T> getNext()
	{
		return this.next;
	}

	public T getElement()
	{
		return this.element;
	}

	public void setElement(T element)
	{
		this.element = element;
	}

	public String toString()
	{
		return element.toString();
	}
}

File: C-examples/C++/stack/Node.cc

#include "Node.h"

/*

	Implementation of the Node class.
*/

Node::Node(char *obj) {
    data = obj;
    next = NULL; // defined to be the value 0
}

void Node::setData(char *obj) {
    data = obj;
}

char *Node::getData() {
    return data;
}

string Node::toString() {
    string *s = new string(data,1);
    return  *s;
}

void Node::setNext(Node *nextPtr) {
    next = nextPtr;
}

Node *Node::getNext() {
    return next;
}

